openshift/fluentd.dc.yaml

```diff
@@ -0,0 +1,308 @@
+              configMap:
+                name: ${APP_NAME}-${INSTANCE}-config
+
+            # fluentd TLS (enable when using TLS)
+            # - name: fluentd-tls
+            #   secret:
+            #     secretName: fluentd-tls
+
+          terminationGracePeriodSeconds: 120
+          containers:
+            - image: docker-registry.default.svc:5000/${NAMESPACE}/${APP_NAME}-custom:${INSTANCE}
+              imagePullPolicy: IfNotPresent
+              livenessProbe:
+                failureThreshold: 3
+                # httpGet:
+                #   path: /
+                #   port: 8080
+                #   scheme: HTTP
+                exec:
+                  command:
+                  - "/bin/sh"
+                  - "-c"
+                  - "[[ $( pgrep ruby | wc -l)  == 2 ]]"
+                initialDelaySeconds: 10
+                periodSeconds: 60
+                successThreshold: 1
+                timeoutSeconds: 1
+              readinessProbe:
+                failureThreshold: 3
+                # httpGet:
+                #   path: /
+                #   port: 8080
+                #   scheme: HTTP
+                exec:
+                  command:
+                  - "/bin/sh"
+                  - "-c"
+                  - "[[ $( pgrep ruby | wc -l)  == 2 ]]"
+                initialDelaySeconds: 10
+                periodSeconds: 10
+                successThreshold: 1
+                timeoutSeconds: 1
+              name: "${APP_NAME}"
+              volumeMounts:
+                # mount fluentd config file
+                - name: fluentd-config-volume
+                  mountPath: /fluentd/etc/fluent.conf
+                  subPath: fluent.conf
+
+                # TLS cert for connecting to fluentd (enable when using TLS)
+                # - mountPath: /fluentd/etc/ssl
+                #   name: fluentd-tls
+              env:
+                - name: ${APP_NAME}-alert-endpoint
+                  valueFrom:
+                    secretKeyRef:
+                      name: ${APP_NAME}-alert
+                      key: alert-endpoint
+
+
+              ports:
+                - containerPort: 2020
+                  protocol: TCP
+                  name: metrics
+
+                - containerPort: 8080
+                  name: http-plugin
+                  protocol: TCP
+      test: false
+      triggers:
+        - type: ConfigChange
+parameters:
+  - name: APP_NAME
+    description: Application name
+    displayName: Application name
+    required: true
+    value: fluentd
+  - name: LABEL_ENV
+    description: Deployment environment
+    displayName: Environment
+    required: true
+    value: dev
+  - name: INSTANCE
+    description: The deployment instance name
+    displayName: Deployment Instance
+    required: true
+    value: master
+  - name: NAMESPACE
+    description: Target namespace reference (i.e. 'ixhmbm-dev')
+    displayName: Target Namespace
+    required: true
+    value: ixhmbm-dev
```

Comment:
Drop this line - we don't want to have a default embedded in here for namespace - this must be explicitly defined on oc cli call.